File: lintcode/easy/classical_binary_search/py/classical_binary_search.py
class Solution:
    # @param {int[]} A an integer array sorted in ascending order
    # @param {int} target an integer
    # @return {int} an integer
    def findPosition(self, A, target):
        import bisect
        
        i = bisect.bisect_left(A, target)
        
        if i < len(A) and A[i] == target:
            return i
            
        return -1

File: lintcode/easy/classical_binary_search/cpp/classical_binary_search.cpp
#include <algorithm>
#include <vector>

class Solution {
public:
    /**
     * @param A an integer array sorted in ascending order
     * @param target an integer
     * @return an integer
     */
    int findPosition(std::vector<int>& A, int target) 
    {
        decltype(A.size()) lo{ 0 };
        decltype(A.size()) hi{ A.size() };
        
        while (lo < hi) 
        {
            const auto mid = lo + (hi - lo) / 2;
            
            if (A[mid] > target) {
                hi = mid;
            } else if (A[mid] < target) {
                lo = mid + 1;
            } else {
                return mid;   
            }
        }
        
        return -1;
    }
};
